src/routing/approved.rs

```diff
@@ -1458,13 +1463,16 @@ impl Approved {
                 sibling.section.elders_info()
             );
 
-            // We can update the sibling knowledge already because we know they also reached consensus
-            // on our `OurKey` so they know our latest key. Need to vote for it first though, to
-            // accumulate the signatures.
-            commands.extend(self.vote(Vote::TheirKnowledge {
-                prefix: *sibling.section.prefix(),
-                key_index: self.section.chain().last_key_index(),
-            })?);
+            if self.section_keys_provider.has_key_share() {
```

Comment:
This is one of the major fixes in this PR. It fixes an issue when demote and split happen at the same time. We still want the demoted node to send the `Sync` message to the new elders. But without this check, the `vote` call would fail (because a demoted node no longer has the secret key share which is needed for voting) which means we would never call the subsequent `send_sync`. This is necessary in case the post-split elders are completely different nodes than the pre-split elders.

Reply:
> We still want the demoted node to send the Sync message to the new elders. But without this check, the vote call would fail (because a demoted node no longer has the secret key share which is needed for voting)

Reply:
The `send_sync` happens outside of the `if` block.